### PR TITLE
feat(nodejs): add license parser to pnpm analyser

### PR DESCRIPTION
**Pull request by oscarbc96**
## Description

pnpm-lock.yaml files do not contain dependency license information.
This PR parses */node_modules/<package_name>/package.json files alongside pnpm-lock.json and identify licenses.

## Related Issues
- Close https://github.com/aquasecurity/trivy/issues/7076
- https://github.com/aquasecurity/trivy/issues/3718

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [x] I've added tests that prove my fix is effective or that my feature works.
- [x] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

**Comment by knqyf263** [comment]
You can reference [this page](https://aquasecurity.github.io/trivy/v0.52/community/contribute/pr/#development) to pass tests. Please feel free to ask if you need assistance.

**Comment by knqyf263** [comment]
Thanks for your contribution. LGTM.
One comment: I want as few test files as possible so we can easily maintain tests in the future. Can you please provide a minimum file structure that meets the case you want to test?

**Comment by oscarbc96** [comment]
Thanks for your feedback! The tests are already consolidated into a minimal structure. The happy path test requires only two dependencies: `ms`, which has no additional dependencies, and `@babel/parser`, which has several indirect dependencies. This allows us to comprehensively test that everything is working correctly.

Due to the nature of these dependencies, it might appear that we have several test data files. However, to keep things simple, everything except the `package.json` file has been removed. This ensures that we maintain the necessary coverage while adhering to the minimal file structure.

**Comment by knqyf263** [comment]
I still feel like `@babel/parser` has too many dependencies. We don't need to use an actual package for unit tests. You can add it to integration tests.

**Comment by oscarbc96** [comment]
You're right. After giving it a second thought, I removed `@babel/parser` since testing indirect dependencies should be handled by the pnpm parser. The pnpm parser already has tests for these dependencies.

**Comment by knqyf263** [comment]
@DmitriyLewen Could you also take a look?

**Review by DmitriyLewen — COMMENTED** [review]
Hello @oscarbc96 
Thanks for your work!

Add comments. Take a look please.

File: pkg/fanal/analyzer/language/nodejs/pnpm/testdata/happy/node_modules/ms
[the remark this answers is not in the document]
Comment (oscarbc96) [reply pkg/fanal/analyzer/language/nodejs/pnpm/testdata/happy/node_modules/ms:1]: Nope, it's not necessary actually.

[the remark this answers is not in the document]
Comment (DmitriyLewen) [reply pkg/fanal/analyzer/language/nodejs/pnpm/testdata/happy/node_modules/ms:1]: remove it please.

**Review by DmitriyLewen — APPROVED** [review]
LGTM

cc. @knqyf263

**Review by knqyf263 — COMMENTED** [review]
I think we should update [an integration test](https://github.com/aquasecurity/trivy/tree/acbec053c985388a26d899e73b4b7f5a6d1fa210/integration/testdata/fixtures/repo/pnpm) similar to [npm](https://github.com/aquasecurity/trivy/tree/acbec053c985388a26d899e73b4b7f5a6d1fa210/integration/testdata/fixtures/repo/npm) (adding `node_modules` so licenses will be detected).

The test case is here.
https://github.com/aquasecurity/trivy/blob/acbec053c985388a26d899e73b4b7f5a6d1fa210/integration/repo_test.go#L105-L112

You can update the golden file with `mage test:updateGolden`.

File: integration/testdata/pnpm-licenses.json.golden
Comment (DmitriyLewen) [inline integration/testdata/pnpm-licenses.json.golden:1]: You don't need to create a new gold file.

You need to update `testdata/pnpm.json.golden` file.

Comment (oscarbc96) [reply integration/testdata/pnpm-licenses.json.golden:1]: I'm not sure if I'm doing this right; the `testdata/pnpm.json.golden` file is not updating. 🤔  
I'm running `mage test:updateGolden`. @DmitriyLewen  Can you help me figure out what might be wrong?

Comment (DmitriyLewen) [reply integration/testdata/pnpm-licenses.json.golden:1]: Sorry, i missed that `list-all-pkgs` flag is not used for `pnpm` testcase.
I added flag and updated golden file in https://github.com/aquasecurity/trivy/pull/7036/commits/602b4d9d70e53fa0e46e57338ff1c532f29a528a

Comment (oscarbc96) [reply integration/testdata/pnpm-licenses.json.golden:1]: Thank you!

**Review by knqyf263 — APPROVED** [review]
Thanks for your great contribution!

**Pull request merged**